### PR TITLE
tenfourfox: Add Aquafox subport and update tenfourfox-devel

### PR DESCRIPTION
**Pull request by BlackBirdLC**
#### Description
Added Aquafox subport, a TenFourFox fork with extra optimizations, and updated tenfourfox-devel to the latest version 20240505.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

**Note:** Testing was conducted by @barracuda156.

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files? **(Testing of building and binary files was conducted by @barracuda156.)**
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: www/tenfourfox/Portfile
Comment (barracuda156) [inline www/tenfourfox/Portfile:12]: It is better not to add unneeded duplicate empty lines.

**Review by barracuda156 — APPROVED** [review]
I have verified both builds locally, so this should be good to merge, once the extra empty line is removed.

File: www/tenfourfox/Portfile
Comment (BlackBirdLC) [reply www/tenfourfox/Portfile:12]: An oversight on my part.

**Comment by barracuda156** [comment]
@BlackBirdLC And please squash commits, since they will be in MacPorts history once merged. Ideally it is better to have one commit for TFF-devel update and another one for aquafox, but no big deal if both are in a single one.

**Comment by barracuda156** [comment]
@reneeotten @herbygillot This only affects PowerPC systems, and I have confirmed it works. Please merge this.

File: www/tenfourfox/Portfile
Comment (reneeotten) [inline www/tenfourfox/Portfile:47]: this is not a "-devel" port so you should use an upstream-tagged / released version.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: www/tenfourfox/Portfile
Comment (barracuda156) [reply www/tenfourfox/Portfile:47]: @BlackBirdLC This is equivalent to v. 1.0.0, right? Then just use that instead of the hash (author repo 1.0.0 v).

Comment (BlackBirdLC) [reply www/tenfourfox/Portfile:47]: @barracuda156 It is with the exception of the nostrip variants of mozcfg, which the tenfourfox Portfile expects to be present.

Comment (barracuda156) [reply www/tenfourfox/Portfile:47]: @BlackBirdLC Got it. Maybe then just add https://github.com/BlackBirdLC/Aquafox/commit/0785667d630acf269af554f538c19da181e5253f commit as a patch on top of v 1.0.0?

Comment (BlackBirdLC) [reply www/tenfourfox/Portfile:47]: @barracuda156 I could create a new release named v1.0.0-0785667 without any binaries attached, or would that be considered cheating?

Comment (barracuda156) [reply www/tenfourfox/Portfile:47]: That would cost me another 3 hrs of compilation with strictly identical result LOL

If you don’t mind, I’t rather prefer keeping 1.0.0 version, just with the patch added, so that we have the same source as before.
Inside `aquafox` subport block add something like:
```
patchfiles-append 0785667d630acf269af554f538c19da181e5253f.patch
```
(Add place the patch inside `files` directory.)

File: www/tenfourfox/Portfile
Comment (reneeotten) [inline www/tenfourfox/Portfile:34]: this isn't right

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by barracuda156** [comment]
@BlackBirdLC Do you want me to handle version and patch? No functional changes, just fix this technically in portfile syntax.

**Comment by BlackBirdLC** [comment]
@barracuda156 Everything should be good now, but my Git tomfoolery caused the pull request to close. Can I just reopen this one, or should I open a new one? You can see the changes I made here: https://github.com/BlackBirdLC/macports-ports/commits/aquafox/

**Comment by barracuda156** [comment]
@BlackBirdLC If it allows to re-open it, please. Otherwise making a new one is fine too.

**Comment by BlackBirdLC** [comment]
@barracuda156 Just a thought: with the commit hash changed to the release tag, the checksums will also need to be updated, as they will be checked against the source pre-patched. Is that correct? I seem to be forgetting one thing after another.

**Comment by barracuda156** [comment]
> @barracuda156 Just a thought: with the commit hash changed to the release tag, the checksums will also need to be updated, as they will be checked against the source pre-patched. Is that correct? I seem to be forgetting one thing after another.

Yes, of course, checksums will differ. They should correspond to v1.0.0.

**Comment by BlackBirdLC** [comment]
> Yes, of course, checksums will differ. They should correspond to v1.0.0.
I've converted the pull request to a draft once again and will work on it. I’ll take my time to review it thoroughly before pushing.

**Comment by BlackBirdLC** [comment]
@barracuda156 I did not anticipate all the changes that were needed after I created the pull request, which led to me hastily trying to push the changes while at the same time being tired and unfamiliar with the workflow. I have just pushed the necessary changes; could you check them before I mark it as ready for review? (to avoid a repetition of the mess I caused yesterday)

File: www/tenfourfox/Portfile
Comment (barracuda156) [inline www/tenfourfox/Portfile:47]: It should be:
```
github.setup        BlackBirdLC Aquafox 1.0.0 v
```
Otherwise the version will be prefixed with `v`, we do not want that.

**Review by barracuda156 — APPROVED** [review]
Assuming checksums are correct and the patch applies (I am away from powerpc machine now), LGTM.

**Comment by barracuda156** [comment]
@reneeotten If you are satisfied with the current state of this, should be good to merge. (Since changes since the last iteration are cosmetical, I am not rebuilding this over again.)

File: www/tenfourfox/Portfile
Comment (reneeotten) [inline www/tenfourfox/Portfile:48]: version is deduced from the "github.setup" line and should not be repeated here.

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Pull request merged**
